### PR TITLE
PIA-1544: Add QR Sign in tvOS validation to iOS

### PR DESCRIPTION
**Pull request by kp-said-rehouni**
**Summary**
This PR adds the validation for the QR Sign in token scanned from tvOS.
It creates a new screen which shows a loading spinner while binding the sign in token with the current api token.

This PR can't be merged until this one is: https://github.com/pia-foss/mobile-ios-library/pull/29

### CODE REVIEW

File: PIA Common/Data/Networking/HTTPClient.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        case accept = "accept"
+        case content_type = "Content-Type"
+    }
+}
+
+protocol HTTPClientType {
+    @available(iOS 13.0.0, *)
+    func makeRequest(request: URLRequest) async throws -> Data
+    func makeRequest(request: URLRequest, completion: @escaping (Result<(Data, URLResponse), ClientError>) -> Void)
+}
```
Comment (kp-laura-sempere) [inline PIA Common/Data/Networking/HTTPClient.swift:44]: Very nice 👍

File: PIA VPN/DashboardViewController.swift [hunk 2]
```diff
@@ -195,6 +196,15 @@ class DashboardViewController: AutolayoutViewController {
 
     }
     
+    private func checkTVOSTokenToBind() {
+        guard let apiToken = Client.providers.accountProvider.apiToken,
+        let token = Client.configuration.tvOSBindToken else { return }
+        
+        guard let viewController = ValidateQRLoginFactory.makeValidateQRLoginViewController(apiToken: apiToken, tvOSBindToken: token) else { return }
+        viewController.modalPresentationStyle = .fullScreen
+        present(viewController, animated: true)
```
Comment (kp-laura-sempere) [inline PIA VPN/DashboardViewController.swift:205]: Question: Why do we present the ValidateQRLoginViewController from here and also from the AppDelegate?

**Review by kp-laura-sempere — APPROVED** [review]
Great job, @kp-said-rehouni 
It looks great! I just posted a question

File: PIA VPN/DashboardViewController.swift (hunk 2 above)
Comment (kp-said-rehouni) [reply PIA VPN/DashboardViewController.swift:205]: Good question!
There are 2 flows that users can go through after opening the app from the camera:
- User is already logged in: The app opens in dashboard and we present the ValidateQRLoginViewController from the AppDelegate.
- User is not logged in: The app opens in Welcome. When the users logs in and gets to Dashboard, we check if there is a tvOS token and if so we present the ValidateQRLoginViewController from the dashboard.

Comment (kp-laura-sempere) [reply PIA VPN/DashboardViewController.swift:205]: OK, got it! Thanks for the clarification. All good 👍

**Pull request merged**